client/components/LivePreviewDemo/IframeDemo/IframeDemo.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+interface Props {
+   sourceCode: string;
+}
+
+export default function IframeDemo({ sourceCode } : Props) {
+  const writeLanguage = (iframe: HTMLIFrameElement) => {
+    if (!iframe) {
+      return;
+    }
+    let doc = iframe.contentDocument;
+    doc.open();
+    doc.write(sourceCode);
+    doc.close();
+    iframe.style.width = '100%';
+    iframe.contentWindow.document.body.style.color = 'silver';
+  };
+
+  return (
+    <iframe frameBorder="0" ref={writeLanguage} title="iframe" />
```

Comment:
It's a little bit weird the way you're using refs here to modify the `iframe`. It works but seems like a hack.

A simpler solution would be to create the `ref` the usual way using `useRef`,  assign it to the iframe and then use the `useEffect` hook to trigger updates when the `sourceCode` variable changes.